File: src/main/java/no/fintlabs/InstanceCleanupService.java
package no.fintlabs;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
@Slf4j
public class InstanceCleanupService {

    @Value("${fint.flyt.instance-service.time-to-keep-instance-in-days:60}")
    private int timeToKeepInstancesInDays;

    private final InstanceService instanceService;

    @Scheduled(initialDelay = 1000, fixedDelay = 86400000)
    public void cleanUp() {
        log.info("Cleaning up instances older than {} days", timeToKeepInstancesInDays);
        instanceService.deleteAllOlderThan(timeToKeepInstancesInDays);
    }
}

File: src/main/kotlin/no/novari/flyt/instance/InstanceCleanupService.kt
package no.novari.flyt.instance

import io.github.oshai.kotlinlogging.KotlinLogging
import org.springframework.beans.factory.annotation.Value
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Service

@Service
class InstanceCleanupService(
    @param:Value("\${novari.flyt.instance-service.time-to-keep-instance-in-days:60}")
    private val timeToKeepInstancesInDays: Int,
    private val instanceService: InstanceService,
) {
    private val log = KotlinLogging.logger {}

    @Scheduled(initialDelay = 30000, fixedDelay = 86400000)
    fun cleanUp() {
        log.atInfo {
            message = "Cleaning up instances older than {} days"
            arguments = arrayOf(timeToKeepInstancesInDays)
        }
        instanceService.deleteAllOlderThan(timeToKeepInstancesInDays)
    }
}
